src/abi/stack.h

```diff
@@ -40,11 +40,43 @@ inline Index stackAlign(Index size) {
 // just local.get it anywhere there.
 inline void
 getStackSpace(Index local, Function* func, Index size, Module& wasm) {
-  auto* stackPointer =
-    GlobalUtils::getGlobalInitializedToImport(wasm, ENV, "STACKTOP");
+
+  // This function expects to see code that was generated by Emscripten, and
+  // further relies on a particular code idiom, specifically, the inclusion
+  // of an exported function "stackSave" with a specific structure, from which
+  // the identity of the global stack pointer (STACKTOP) may be inferred.
+  //
+  // We expect to find the module structured as follows:
+  //
+  //(module
+  //  ...
+  //  (export "stackSave" (func $stackSave))
+  //  ...
+  //  (func $stackSave (; 410 ;) (; has Stack IR ;) (result i32)
+  //    (global.get $STACKTOP)
+  //  )
+  //  ...
+  //)
+
+  Global* stackPointer = nullptr;
+  Export* stackSaveFunctionExport = wasm.getExportOrNull("stackSave");
+  if (stackSaveFunctionExport) {
+    Function* stackSaveFunction =
+      wasm.getFunctionOrNull(stackSaveFunctionExport->value);
+    if (stackSaveFunction) {
+      Expression* functionBody = stackSaveFunction->body;
+      if (functionBody) {
```

Comment:
no need to check if the body is null, as it must exist. can add an `assert(!stackSaveFunction->imported());` just to be sure.